<launch>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find robotic_arm)/config/robotic_arm.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	output="screen" ns="/robotic_arm" args="joint_state_controller
					  0_shoulder_LR_position_controller
					  1_shoulder_FB_position_controller
					  2_elbow_FB_position_controller
					  3_elbow_R_position_controller
					  4_wrist_P_position_controller
					  5_wrist_Y_position_controller
            6_hand_left_position_controller
            7_hand_right_position_controller
            wheel_LF_velocity_controller
            wheel_RF_velocity_controller
            wheel_LB_velocity_controller
            wheel_RB_velocity_controller
            "/> 

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
	respawn="false" output="screen">
    <remap from="/joint_states" to="/robotic_arm/joint_states" />
  </node>

  <!-- <node name="serial_publisher" pkg="robotic_arm" type="serial_publisher.py" respawn="false" output="screen"/> -->

</launch>
